code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm

```diff
@@ -0,0 +1,448 @@
+		if("outputKW")
+			power_format_multi_output = 1 KW
+		if("outputMW")
+			power_format_multi_output = 1 MW
+		if("outputGW")
+			power_format_multi_output = 1 GW
+
+/obj/machinery/power/transmission_laser/process()
+	max_grid_load = get_surplus()
+	input_available = get_surplus()
+	if(stat & BROKEN)
+		return
+
+	if(total_energy >= announcement_threshold)
+		send_ptl_announcement()
+
+	var/last_disp = return_charge()
+	var/last_chrg = inputting
+	var/last_fire = firing
+
+	if(last_disp != return_charge() || last_chrg != inputting || last_fire != firing)
+		update_icon()
+
+	if(powernet && input_attempt && turned_on)
+		input_pulling = min(input_available , input_number * power_format_multi)
+
+		if(inputting)
+			if(input_pulling > 0)
+				consume_direct_power(input_pulling)
+				charge += input_pulling
+			else
+				inputting = FALSE
+		else
+			if(input_attempt && input_pulling > 0)
+				inputting = TRUE
+	else
+		inputting = FALSE
+
+	if(charge < MINIMUM_POWER)
+		firing = FALSE
+		output_level = 0
+		destroy_lasers()
+		return
+
+	if(!firing)
+		return
+
+	output_level = min(charge, output_number * power_format_multi_output)
+	if(!length(laser_effects))
+		setup_lasers()
+
+	if(length(blocked_objects))
+		var/atom/listed_atom = blocked_objects[1]
+		if(prob(max((abs(output_level) * 0.1) / 500 KW, 1))) ///increased by a single % per 500 KW
+			listed_atom.visible_message("<span class='danger'>[listed_atom] is melted away by the [src]!<span/>")
+			blocked_objects -= listed_atom
+			qdel(listed_atom)
+
+	else
+		sell_power(output_level * WATT_TICK_TO_JOULE)
+
+	charge -= output_level
+
+//// Selling defines are here
+#define MINIMUM_BAR 1
+#define PROCESS_CAP (20 - MINIMUM_BAR)
+
+#define A1_CURVE 20
+
+/obj/machinery/power/transmission_laser/proc/sell_power(joules)
+	var/mega_joules = joules / (1 MW)
+
+	var/generated_cash = (2 * mega_joules * PROCESS_CAP) / ((2 * mega_joules) + (PROCESS_CAP * A1_CURVE))
+	generated_cash += (4 * mega_joules * MINIMUM_BAR) / (4 * mega_joules + MINIMUM_BAR)
+	generated_cash = round(generated_cash)
+	if(generated_cash < 0)
+		return
+
+	total_energy += mega_joules
+	total_earnings += generated_cash
+	generated_cash += unsent_earnings
+	unsent_earnings = generated_cash
+
+	var/datum/money_account/engineering_bank_account = GLOB.station_money_database.get_account_by_department(DEPARTMENT_ENGINEERING)
+	var/datum/money_account/cargo_bank_account = GLOB.station_money_database.get_account_by_department(DEPARTMENT_SUPPLY)
+
+	var/medium_cut = generated_cash * 0.25
+	var/high_cut = generated_cash * 0.75
+
+	cargo_bank_account.deposit_credits(medium_cut, "Transmission Laser Payout")
+	unsent_earnings -= medium_cut
+
+	engineering_bank_account.deposit_credits(high_cut, "Transmission Laser Payout")
+	unsent_earnings -= high_cut
+
+#undef A1_CURVE
+#undef PROCESS_CAP
+#undef MINIMUM_BAR
+
+// Beam related procs
+
+/obj/machinery/power/transmission_laser/proc/setup_lasers()
+	/// This is why we set the range we did
```

Comment:
What does this comment mean? 
```suggestion
	// This is why we set the range we did
```

Reply:
It really isn't required. Just means this is where we use the range variable from what I can see, but that's quite self explanatory.